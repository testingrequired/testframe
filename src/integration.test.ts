import { run, config, middleware } from "./index";

const {
  starter,
  specSyntax,
  matchTestFiles,
  random,
  mock,
  fixture
} = middleware;

describe("Integration", () => {
  // @ts-ignore
  let oldExit: any;
  // @ts-ignore
  let oldDateNow: any;

  let logs: Array<Array<string>>;

  beforeEach(() => {
    oldExit = process.exit;
    oldDateNow = Date.now;

    (process.exit as any) = jest.fn();

    logs = [];

    (global as any).console = new Proxy(console, {
      get: (target, prop: string) => {
        return (...args: Array<string>) => {
          logs.push(args);
        };
      }
    });

    Date.now = jest
      .fn()
      .mockReturnValueOnce(100)
      .mockReturnValueOnce(200);
  });

  it("should work", async () => {
    await run(
      config(
        starter,
        matchTestFiles("./tests/**/*.test.js"),
        specSyntax,
        random,
        mock,
        fixture("testFixture", () => "expected fixture value")
      )
    );

    expect(logs.join("\n")).toMatchSnapshot();
  });
});
