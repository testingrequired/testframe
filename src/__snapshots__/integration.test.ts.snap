// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration should work 1`] = `
"testframe

Args: {\\"_\\":[]}

Test Globals: assert, console, random, mock, when, verify, replace

Test File Paths: tests/assert.test.js, tests/async.test.js, tests/beforeEach.test.js, tests/mock.test.js, tests/skip.test.js, tests/todo.test.js

1..12
ok 1 assert should work
ok 2 async should work for before each
ok 3 async should work for test
ok 4 beforeEach should initialize value
ok 5 beforeEach increment should increment value
ok 6 beforeEach decrement should decrement value
ok 7 mock should work
ok 8 skip single test # skip
ok 9 skip block of test test 1 # skip
ok 10 skip block of test test 2 # skip
ok 11 skip block of test nested also works test 3 # skip
not ok 12 A test that needs done


Logs

- warn: \\"Warning: testdouble.js - td.verify - test double was both stubbed and verified with arguments (\\\\\\"Symbol()\\\\\\"), which is redundant and probably unnecessary. (see: https://github.com/testdouble/testdouble.js/blob/master/docs/B-frequently-asked-questions.md#why-shouldnt-i-call-both-tdwhen-and-tdverify-for-a-single-interaction-with-a-test-double )\\"


"
`;
